src/decode.c

```diff
@@ -407,19 +408,19 @@ static inline bool op_load(rv_insn_t *ir, const uint32_t insn)
     /* dispatch from funct3 field */
     switch (decode_funct3(insn)) {
     case 0: /* LB: Load Byte */
-        ir->opcode = rv_insn_lb;
+        ir->opfunc = (rvop_t) &rvop_lb;
```

Comment:
The assignment for `opfunc` should be inside basic block generation.

Reply:
When used in conjunction with computed-goto, we can assign "opfunc" to a function's corresponding label rather than a function pointer. As a result, we may reuse the compiled code and even get rid of the cost of early function calls.

Reply:
The dispatch_table and label are defined in function emulate, I have no idea to assign a function's corresponding label to "opfunc" inside basic block generation.

Reply:
No problem. I will discuss [code-copying](http://www.sable.mcgill.ca/publications/papers/2008-2/paper.pdf) approach later. At present, you can proceed the minimal changes on top of the latest `master` branch. The separation among different module is vital because we are about to implement JIT compilation sooner or later.